### ISSUE TITLE
Peer review and reproducibility

### ISSUE THREAD

**Issue by indeous** [risenyard/Machine_Learning_Practice#1]
### **Computation**

**Tops:**
- The use of additional indices to increase the dimensionality of data is appreciated. An evaluation of how much the use of indices along with band statistics effect the accuracy of the model can be evaluated to justify this.
- Good justification for choosing Random Forest model with references.
- The comparison of Random Forest with and without Dask is relevant by using the same algorithm. 
- Reading the files using Dask is a plus point. This speeds up the computation of reading the file.
- Using a workaround for HyperBandSearchCV is very interesting. It does reduce the computation time.
- Explanation of memory profiling results and Dask dashboard are interesting and helpful in understanding how Dask works. 

**Tips:**
- Reading the dataset took more time than expected. This maybe due to reading all bands, some bands like aerosols and clouds may not be of use in landcover classification. Additionally, reading the bands as `int16 `excludes the decimal points of the band values. 
```
                    bands = src.read()
                    # Transfer the uint16 to int16
                    bands = bands.astype(np.int16)
```
   This is crucial as the quality of the dataset significantly determines how well a machine learning model can be trained. Float datatype can be used instead. Moreover, after reading all bands, this data of the image is not used. Only band values of the centroid pixel are used. 
```
                    lon, lat = src.xy(src.width // 2, src.height // 2)
                    row, col = src.index(lon, lat)
                    bands = bands[:, row, col]
                    ...
                    mean_val = np.mean(bands)
```
   Only centroid pixel may not represent statistics of the whole image. Statistics are then calculated across all band values of one pixel, instead of within each band of the image. Using this data could negatively effect the model training. Indices created with 'float' values can be helpful in training a model. The indices and statistics columns have been left out as of now.
- The use of k-fold cross validation experimented is interesting. However, by splitting into 10 folds, each training set has 90% samples, and test set has 10% samples. Training a RF model on 90% of data may overfit. And checking accuracy on only 10% of data is not optimal as it can give low accuracy. Usually k-fold is used when the number of training samples are too few. In our case, as there are 27000 samples, it may not be required. 
- Apart from this, scikit-learn RF algorithm takes "bootstrap" parameter as "True" by default. Which means, the training data is already bootstrapped with random selection of samples for building each decision-tree and scoring using out-of-bag samples. Therefore, using k-fold cross validation in a bootstrapped RF algorithm may not be required. 

### **Presentation**

**Tops:**
- Explanation of memory profiling results and Dask dashboard are presented very well.

**Tips:**
- Reflection on results and accuracy of best model from HPO can add value to reading of the notebook.

### **Reproducibility**

Overall, the notebooks are well written and can be reproduced to learn the implementation of Dask in RF machine learning algorithm. The link to EuroSAT dataset is posted for anyone without access to the directory. 

I did not run Grid search in the non-Dask version as it takes too long, rest of the code in the non-DASK version runs well.

In the Dask version, the following part of the code causes problems on CRIB. However, on a personal system with latest version of `dask_ml.model_selection` imported, this code runs well.
```
pa_mp = accuracy_score(y_test, pred_labels, normalize=False)
```
It throws the following TypeError, 
`Expected sequence or array-like, got <class 'dask.dataframe.core.Series'>`

Maybe this happens as a Dask object `y_test `is being passed to non-dask function. It may have to be computed before calculating the accuracy.
